### PR TITLE
Reactify mute remote participant

### PR DESCRIPTION
**Pull request by yanas**

### CODE REVIEW

**Comment by yanas** [comment]
NOTE: This is a valid test fail and I need to retest and have a look.

File: react/features/base/participants/middleware.js [hunk 1]
```diff
@@ -37,30 +39,15 @@ MiddlewareRegistry.register(store => next => action => {
...
         break;
 
-    case MUTE_REMOTE_PARTICIPANT:
-        if (typeof APP !== 'undefined') {
-            APP.UI.messageHandler.openTwoButtonDialog({
-                titleKey: 'dialog.muteParticipantTitle',
-                msgString:
-                    '<div data-i18n="dialog.muteParticipantBody"></div>',
-                leftButtonKey: 'dialog.muteParticipantButton',
-                dontShowAgain: {
```
Comment (virtuacoplenny) [inline react/features/base/participants/middleware.js]: The removal of this dontShowAgain was intentional? That saves a lot of work if it's intentional. Maybe I've already said the same exact thing previously.

File: react/features/base/participants/middleware.js [hunk 2]
```diff
@@ -37,30 +39,15 @@ MiddlewareRegistry.register(store => next => action => {
         break;
 
     case KICK_PARTICIPANT:
-        if (typeof APP !== 'undefined') {
-            APP.UI.emitEvent(UIEvents.USER_KICKED, action.id);
```
Comment (virtuacoplenny) [inline react/features/base/participants/middleware.js]: I think you can remove the USER_KICKED event and listener.

File: react/features/base/participants/middleware.js [hunk 3]
```diff
@@ -37,30 +39,15 @@ MiddlewareRegistry.register(store => next => action => {
...
-                    textKey: 'dialog.doNotShowMessageAgain',
-                    checked: true,
-                    buttonValues: [ true ]
-                },
-                submitFunction: () => {
-                    APP.UI.emitEvent(UIEvents.REMOTE_AUDIO_MUTED, action.id);
-                }
-            });
-        }
+    case MUTE_REMOTE_PARTICIPANT: {
```
Comment (virtuacoplenny) [inline react/features/base/participants/middleware.js]: Silly style nit: The rest of the switch doesn't use curly braces so maybe this line shouldn't either. Curly braces would be needed for scoping variable declarations.

**Comment by virtuacoplenny** [comment]
Looks good to me. The test fail is at least a test-side failure and not a code execution failure. The tests were modifying local storage so the dialog would not display. https://github.com/jitsi/jitsi-meet-torture/blob/3f06a6ed4dc727a3448144101dd205b4479c74b7/src/test/java/org/jitsi/meet/test/ConferenceFixture.java#L422.

**Comment by virtuacoplenny** [comment]
Jenkins test this please.

**Comment by virtuacoplenny** [comment]
Tests are fixed with a change like https://github.com/jitsi/jitsi-meet-torture/pull/101. 

On a side thought: does the mute confirmation dialog even need to exist?

File: react/features/base/participants/middleware.js (hunk 1 above)
Comment (yanas) [reply react/features/base/participants/middleware.js]: Yes, it was intentional. I started advocating for not adding such checks in general.

File: react/features/base/participants/middleware.js (hunk 2 above)
Comment (yanas) [reply react/features/base/participants/middleware.js]: Indeed, I've noticed that yesterday and will remove those for kick and mute.

**Comment by yanas** [comment]
We don't want users to accidentally mute participants, especially in a scenario where everyone has moderator rights. Adding an extra barrier makes sense to prevent bug reports around people being muted somehow inexplicably.

**Pull request merged**